test/dirichlet.jl

```diff
@@ -127,3 +128,54 @@ end
     @test entropy(Dirichlet(N, 1)) ≈ -loggamma(N)
     @test entropy(Dirichlet(ones(N))) ≈ -loggamma(N)
 end
+
+@testset "Dirichlet differentiation $n" for n in (2, 10)
+    alpha = rand(n)
+    Δalpha = randn(n)
+    d, ∂d = ChainRulesCore.frule((nothing, Δalpha), Dirichlet, alpha)
+    ChainRulesTestUtils.test_frule(Dirichlet ⊢ ChainRulesCore.NoTangent(), alpha ⊢ Δalpha)
```

Comment:
This causes errors with finite differencing since `alpha` can become negative (https://github.com/JuliaStats/Distributions.jl/runs/6145567730?check_suite_focus=true#step:6:503). I guess using `forward_fdm` instead of `central_fdm` should fix the problem (https://github.com/JuliaDiff/FiniteDifferences.jl#dealing-with-singularities).

```suggestion
    ChainRulesTestUtils.test_frule(Dirichlet ⊢ ChainRulesCore.NoTangent(), alpha ⊢ Δalpha; fdm=FiniteDifferences.forward_fdm(5, 1))
```

Reply:
Also `NoTangent` should be the default here (https://github.com/JuliaDiff/ChainRulesTestUtils.jl/blob/5f523e963c0d86018074beadf9019c2147975c46/src/rand_tangent.jl#L55):
```suggestion
    ChainRulesTestUtils.test_frule(Dirichlet, alpha ⊢ Δalpha; fdm=FiniteDifferences.forward_fdm(5, 1))
```